adobe/react-spectrum
# Fix correctly handle IME Input in Combobox

related to https://github.com/nextui-org/nextui/issues/2121

This PR addresses a bug in the combobox component where inputs from an IME were being incorrectly committed before the user completed their text entry.

## ✅ Pull Request Checklist:


### before

https://github.com/adobe/react-spectrum/assets/76232929/e92e8dcb-490c-4b14-bcb0-d5cb66c545e9

### after

https://github.com/adobe/react-spectrum/assets/76232929/39722d5b-0f72-4e0e-82b2-07ce68bd4e0f

## 📝 Test Instructions:


### Setup for MacOS
Add the Japanese input method
- Settings > Keyboard > Input Sources > Add a language > Japanese

### Test Steps
1. Choose Hiragana in the menu bar.
2. Navigate to http://localhost:9003/?path=/story/react-aria-components--combo-box-ime-example&providerSwitcher-express=false&strict=true
3. Type `nihon` followed by the enter key. This should result in "にほん" or "日本".
4. Continue typing `go`, and the combobox should suggest "にほんご". Press enter.
5. Previously, the initial input "にほん" would be cleared, leaving only "ご". After this fix, "にほんご" should be correctly inputted without clearing the previous characters.



## 🧢 Your Project:


https://github.com/nextui-org/nextui
